### PR TITLE
[doc] Add some documentation in order to start integrating Sirius Components

### PR DESCRIPTION
**Pull request by sbegaudeau**
Signed-off-by: Stéphane Bégaudeau <stephane.begaudeau@obeo.fr>

### CODE REVIEW

File: doc/reference/reference.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+A significant part of this code could be moved to Sirius Web in the future.
+
+The main project of our GraphQL layer is `sirius-components-graphql` which contains the integration of GraphQL Java to both the HTTP API and WebSockets API of Spring using respectively the `GraphQLController` and `GraphQLWebSocketHandler`.
+If you want to debug how a GraphQL request is processed on the HTTP API or the WebSocket API, these are the relevant entry point.
+
+===== Schema
+
+The GraphQL schema of Sirius Components is defined using several distinct `.graphqls` files.
+This allow an application to only integrate the parts that it requires.
+The only required part of the schema is located in `sirius-components-collaborative/src/main/resources/schema/code.graphqls`.
```
Comment (gcoutable) [inline doc/reference/reference.adoc:62]: `core.graphqls`

File: doc/reference/reference.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+This file contains the code concepts of our GraphQL schema and is required by all the other schema files.
```
Comment (gcoutable) [inline doc/reference/reference.adoc:63]: the _core_ concepts of ...

**Review by gcoutable — CHANGES_REQUESTED** [review]

File: doc/reference/reference.adoc (hunk 1 above)
Comment (sbegaudeau) [reply doc/reference/reference.adoc]: Done

File: doc/reference/reference.adoc (hunk 2 above)
Comment (sbegaudeau) [reply doc/reference/reference.adoc]: Done

**Review by gcoutable — APPROVED** [review]

**Pull request merged**